compiler/rustc_const_eval/src/interpret/operand.rs

```diff
@@ -670,19 +670,24 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
 
         trace!("eval_place_to_op: got {:?}", op);
         // Sanity-check the type we ended up with.
-        debug_assert!(
-            mir_assign_valid_types(
```

Comment:
We have pretty much the same check in `eval_place` in `place.rs`, can you fix that one as well?